library/private/options_tools.h.in

```diff
@@ -10,14 +10,45 @@ namespace f3d
 {
 namespace options_tools
 {
-// TODO expose parse methods in options API ?
 
 //----------------------------------------------------------------------------
 /**
- * Not implemented generic templated parse method
+ * A std::string_view trim function
  */
+std::string trim(std::string_view strv)
+{
+  constexpr std::string_view spaces = " \t\r\v\n";
+  strv.remove_prefix(std::min(strv.find_first_not_of(spaces), strv.size()));
+  strv.remove_suffix(std::min(strv.size() - strv.find_last_not_of(spaces) - 1, strv.size()));
+  return std::string(strv);
```

Comment:
I think there's an implicit constructor doing the conversion for you.
```suggestion
  return strv;
```

Reply:
` error: could not convert ‘strv’ from ‘std::string_view’ {aka ‘std::basic_string_view<char>’} to ‘std::string’`:

https://stackoverflow.com/questions/47525238/why-is-there-no-implicit-conversion-from-stdstring-view-to-stdstring